src/client/reducers/audioReducer.js

```diff
@@ -0,0 +1,64 @@
+import { CREATE_AUDIO_DATA, PLAY_PAUSE_SOUND_FROM_FILE, CREATE_STREAME_DATA, START_MUTE_STREAME_AUDIO, MERGE_CANVAS_WIDTH } from '../actions/types';
+
+var context = new (window.AudioContext || window.webkitAudioContext)();    
+var analyser = context.createAnalyser();
+
+
+const initialState={  
+    //graphic canvas 
+     widthCanvas: 500,
+     heightCanvas: 150, 
+    //audio from file
+     trackname: undefined,     
+     tracktype: undefined,
+     tracksize: undefined,     
+     audiocontext: context,
+     analyser: analyser,
+     audioFile: undefined,
+     audioFromFile: undefined,
+     audioFromFileSource: undefined,
+    //from stream
+     audioStream: undefined,
+     streamSource: undefined,     
+     playpausestate: false,
+     startMuteState: false
+};
+
+
+export default function (state=initialState, action){
+    
+    switch(action.type){
+        case CREATE_AUDIO_DATA:                         
+            return {
+                ...state,
+                trackname: action.payload.name,
```

Comment:
You can destruct props from action.payload

Reply:
done